lib/IonKey.ts

```diff
@@ -1,12 +1,18 @@
 import * as Ed25519 from '@noble/ed25519';
 import * as Secp256k1 from '@noble/secp256k1';
-import InputValidator from './InputValidator';
-import IonPublicKeyModel from './models/IonPublicKeyModel';
-import IonPublicKeyPurpose from './enums/IonPublicKeyPurpose';
-import JwkEd25519 from './models/JwkEd25519';
-import JwkEs256k from './models/JwkEs256k';
+import InputValidator from './InputValidator.js';
+import IonPublicKeyModel from './models/IonPublicKeyModel.js';
+import IonPublicKeyPurpose from './enums/IonPublicKeyPurpose.js';
+import JwkEd25519 from './models/JwkEd25519.js';
+import JwkEs256k from './models/JwkEs256k.js';
 import { base64url } from 'multiformats/bases/base64';
 
+// NOTE: @noble/secp256k1 requires globalThis.crypto polyfill for node.js <=18: https://github.com/paulmillr/noble-secp256k1/blob/main/README.md#usage
+// Remove when we move off of node.js v18 to v20, earliest possible time would be Oct 2023: https://github.com/nodejs/release#release-schedule
+import { webcrypto } from 'node:crypto';
+// @ts-ignore
+if (!globalThis.crypto) { globalThis.crypto = webcrypto; }
+
```

Comment:
Shouldn't users be doing this in their own projects? I feel like this should be part of our README, but should not be part of the code of this package. It's probably going to lead to compat issues for users who use this project to have this here. I would also make sure that this very same code isn't in any of the associated web5 SDKs.

Reply:
@shamilovtim, thanks for the call outs, when I tested the same change in `dwn-sdk-js`, the code works fine in browser environment, I didn't dig why, but moved on knowing that it works. But I have made the following changes in this PR according to your recommendations:

1. Changed to dynamic import, and moved the dynamic import to tests (as they are run in node.js thus need it).
1. Updated `README.md`.

Reply:
I guess this particular line is not a huge deal but my main concern is that we're altering globals in people's projects without consent. In my experience working with some of the libs in this space, this causes more problems than it solves. Debugging node_modules that do stuff like this can be extremely frustrating, and much more difficult than copy/pasting a line in a README file. In fact in RN right now we're dealing with problems caused by deps that do this very kind of thing. Thoughts?